sui_programmability/adapter/src/adapter.rs

```diff
@@ -405,7 +409,33 @@ pub fn store_package_and_init_modules<
 
     // wrap the modules in an object, write it to the store
     // The call to unwrap() will go away once we remove address owner from Immutable objects.
-    let package_object = Object::new_package(modules, ctx.digest());
+
+    // Must populate commitments to deps
+    let mut deps = BTreeSet::new();
+    for m in &modules {
+        let mut g: BTreeSet<ObjectID> = m
+            .module_handles
+            .iter()
+            .filter_map(|j| {
+                let r = ObjectID::from(*m.address_identifier_at(j.address));
+                if r == package_id {
+                    None
+                } else {
+                    Some(r)
+                }
+            })
+            .collect();
+        deps.append(&mut g);
+    }
```

Comment:
All the dependent packages are already in the input_objects, we should be able to pass them in from `execution_engine::execute_transaction`.

Reply:
Great, in that case, I'll simplify.